components/github/github.app.mjs

```diff
@@ -1,529 +1,536 @@
-
-      return response.data;
-    },
-    async getCommits({
-      repoFullname, ...data
-    }) {
-      const { data: commits } = await this._client().request(`GET /repos/${repoFullname}/commits`, {
-        ...data,
-      });
-      return commits;
-    },
-    async getBranches({
-      repoFullname, ...data
-    }) {
-      const { data: branches } = await this._client().request(`GET /repos/${repoFullname}/branches`, {
-        ...data,
-      });
-      return branches;
-    },
-    async getProjectCards({
-      columnId, ...data
-    }) {
-      const { data: cards } = await this._client().request(`GET /projects/columns/${columnId}/cards`, {
-        ...data,
-      });
-      return cards;
-    },
-    async createOrUpdateFileContent({
-      repoFullname,
-      path,
-      fileContent,
-      commitMessage,
-      branch = null,
-    }) {
-      const data = {
-        message: commitMessage,
-        content: Buffer.from(fileContent).toString("base64"),
-      };
-      const fileExists = await this._makeRequest({
-        path: `/repos/${repoFullname}/contents/${path}`,
-        validateStatus: () => true,
-      });
-      if (fileExists.sha) {
-        console.log("File exists, overwriting.");
-        data.sha = fileExists.sha;
-      }
-      if (branch) {
-        data.branch = branch;
-      }
-      return this._makeRequest({
-        path: `/repos/${repoFullname}/contents/${path}`,
-        method: "put",
-        data: data,
-      });
-    },
-  },
-};
+import { Octokit } from "@octokit/core";
+import { paginateRest } from "@octokit/plugin-paginate-rest";
+import queries from "./common/queries.mjs";
+import { axios } from "@pipedream/platform";
+import { ConfigurationError } from "@pipedream/platform";
+
+const CustomOctokit = Octokit.plugin(paginateRest);
+
+export default {
+  type: "app",
+  app: "github",
+  propDefinitions: {
+    orgName: {
+      label: "Organization",
+      description: "The name of the Github organization. The name is not case sensitive.",
+      type: "string",
+      async options() {
+        const organizations = await this.getOrganizations();
+
+        return organizations.map((organization) => organization.login);
+      },
+    },
+    repoFullname: {
+      label: "Repository",
+      description: "The name of the repository. The name is not case sensitive",
+      type: "string",
+      async options({ org }) {
+        const repositories = await this.getRepos({
+          org,
+        });
+
+        return repositories.map((repository) => repository.full_name);
+      },
+    },
+    repoOrg: {
+      label: "Organization Repository",
+      description: "The repository in a organization",
+      type: "string",
+      async options({ org }) {
+        const repositories = await this.getOrgRepos({
+          org,
+        });
+        return repositories.map((repository) => repository.full_name.split("/")[1]);

```

Comment:
Optional: I noticed this throws an error if `org` is not specified, so we could display the appropriate message.
```suggestion
      async options({ org }) {
      if (!org) {
          throw new ConfigurationError("Must specify `org` to display repository options.");
        }
        const repositories = await this.getOrgRepos({
          org,
        });
        return repositories.map((repository) => repository.full_name.split("/")[1]);

```